Signal/src/ViewControllers/ConversationView/ConversationViewItem.m

```diff
@@ -336,70 +331,81 @@ - (DisplayableText *)displayableTextForInteractionId:(NSString *)interactionId
     return displayableText;
 }
 
+- (nullable TSAttachment *)firstAttachmentIfAnyOfMessage:(TSMessage *)message
+{
+    if (message.attachmentIds.count == 0) {
+        return nil;
+    }
+    NSString *_Nullable attachmentId = message.attachmentIds.firstObject;
+    if (attachmentId.length == 0) {
+        return nil;
+    }
+    return [TSAttachment fetchObjectWithUniqueID:attachmentId];
+}
+
 - (void)ensureViewState
 {
-    OWSAssert([self.interaction isKindOfClass:[TSMessage class]]);
+    OWSAssert([self.interaction isKindOfClass:[TSOutgoingMessage class]] ||
+        [self.interaction isKindOfClass:[TSIncomingMessage class]]);
 
     if (self.hasViewState) {
         return;
     }
     self.hasViewState = YES;
 
-    TSMessage *interaction = (TSMessage *)self.interaction;
```

Comment:
It turns out the `TSMessage.body` can be non-nil (Empty string: `@""`) for incoming attachment messages.  So we need to check for an attachment first.